src/compiler/diagnosticMessages.json

```diff
@@ -1546,6 +1546,52 @@
         "code": 2510
     },
 
+    "JSX element attributes type '{0}' must be an object type.": {
+        "category": "Error",
+        "code": 2600
+    },
+    "The return type of a JSX element constructor must return an object type.": {
+        "category": "Error",
+        "code": 2601
+    },
+    "The global type 'JSX.Element' must exist when using JSX.": {
+        "category": "Error",
+        "code": 2602
+    },
+    "Property '{0}' in type '{1}' is not assignable to type '{2}'": {
+        "category": "Error",
+        "code": 2603
+    },
+    "JSX element '{0}' is not a constructor function.": {

```

Comment:
I wonder if we could just get away with the following message and get rid of this one. I could see it either way.

Also, should these be `"JSX element '{0}' does not have a construct signature"`?